In [1]:
import numpy as np
import pandas as pd

from sklearn.model_selection import train_test_split

from sklearn.ensemble import RandomForestRegressor

from sklearn.preprocessing import RobustScaler 

from sklearn.metrics import mean_squared_error

# 🏆 2do diamonds
##  RSME = 548.80
## Score competi = 552

In [2]:
diamonds_1 = pd.read_csv('./data/diamonds_1.csv')
diamonds_2 = pd.read_csv('./data/diamonds_2.csv')
diamonds_3 = pd.read_csv('./data/diamonds_3.csv')
diamonds_4 = pd.read_csv('./data/diamonds_4.csv')
diamonds_5 = pd.read_csv('./data/diamonds_5.csv')
diamonds_6 = pd.read_csv('./data/diamonds_6.csv')
diamonds_7 = pd.read_csv('./data/diamonds_7.csv')

In [3]:
print(diamonds_1.shape)
print(diamonds_2.shape)
print(diamonds_3.shape)
print(diamonds_4.shape)
print(diamonds_5.shape)
print(diamonds_6.shape)
print(diamonds_7.shape)

(40439, 26)
(40439, 9)
(40439, 33)
(40439, 7)
(40439, 21)
(40439, 19)
(40439, 10)


In [4]:
X_1 = diamonds_1.drop('price', axis=1)
y_1 = diamonds_1['price']

X_2 = diamonds_2.drop('price', axis=1)
y_2 = diamonds_2['price']

X_3 = diamonds_3.drop('price', axis=1)
y_3 = diamonds_3['price']

X_4 = diamonds_4.drop('price', axis=1)
y_4 = diamonds_4['price']

X_5 = diamonds_5.drop('price', axis=1)
y_5 = diamonds_5['price']

X_6 = diamonds_6.drop('price', axis=1)
y_6 = diamonds_6['price']

X_7 = diamonds_7.drop('price', axis=1)
y_7 = diamonds_7['price']

# ⚠️ NO SCALER ⚠️

In [106]:
# X_1_train, X_1_test, y_1_train, y_1_test = train_test_split(X_1, y_1, test_size=0.2, random_state=42)

X_2_train, X_2_test, y_2_train, y_2_test = train_test_split(X_2, y_2, test_size=0.2, random_state=42)

# X_3_train, X_3_test, y_3_train, y_3_test = train_test_split(X_3, y_3, test_size=0.2, random_state=42)

# X_4_train, X_4_test, y_4_train, y_4_test = train_test_split(X_4, y_4, test_size=0.2, random_state=42)

# X_5_train, X_5_test, y_5_train, y_5_test = train_test_split(X_5, y_5, test_size=0.2, random_state=42)

# X_6_train, X_6_test, y_6_train, y_6_test = train_test_split(X_6, y_6, test_size=0.2, random_state=42)

# X_7_train, X_7_test, y_7_train, y_7_test = train_test_split(X_7, y_7, test_size=0.2, random_state=42)

In [107]:
model = RandomForestRegressor()

In [108]:
# model.fit(X_1_train, y_1_train)
model.fit(X_2_train, y_2_train)
# model.fit(X_3_train, y_3_train)
# model.fit(X_4_train, y_4_train)
# model.fit(X_5_train, y_5_train)
# model.fit(X_6_train, y_6_train)
# model.fit(X_7_train, y_7_train)

RandomForestRegressor()

In [109]:
# pred_1 = model.predict(X_1_test)
pred_2 = model.predict(X_2_test)
# pred_3 = model.predict(X_3_test)
# pred_4 = model.predict(X_4_test)
# pred_5 = model.predict(X_5_test)
# pred_6 = model.predict(X_6_test)
# pred_7 = model.predict(X_7_test)

In [33]:
# rmse_1 = mean_squared_error(y_1_test, pred_1, squared=False)
# rmse_2 = mean_squared_error(y_2_test, pred_2, squared=False)
# rmse_3 = mean_squared_error(y_3_test, pred_3, squared=False)
# rmse_4 = mean_squared_error(y_4_test, pred_4, squared=False)
# rmse_5 = mean_squared_error(y_5_test, pred_5, squared=False)
# rmse_6 = mean_squared_error(y_6_test, pred_6, squared=False)
rmse_7 = mean_squared_error(y_7_test, pred_7, squared=False)

C:\Users\tete1\miniconda3\envs\jupyter_env\lib\site-packages\sklearn\metrics\_regression.py:483: FutureWarning: 'squared' is deprecated in version 1.4 and will be removed in 1.6. To calculate the root mean squared error, use the function'root_mean_squared_error'.
  warnings.warn(


In [34]:
print(rmse_1)
print(rmse_2)
print(rmse_3)
print(rmse_4)
print(rmse_5)
print(rmse_6)
print(rmse_7)

631.3863454517286
548.8069601193275
642.8155442636667
559.5984441464017
619.070609363742
560.1013301400569
551.811370099763


## 🏅 Winner: 2do diamonds RSME = 548.80

# ⚠️ ROBUST SCALER ⚠️

In [78]:
robust_scaler = RobustScaler()

In [79]:
# X_1_rob = robust_scaler.fit_transform(X_1)
# X_2_rob = robust_scaler.fit_transform(X_2)
# X_3_rob = robust_scaler.fit_transform(X_3)
# X_4_rob = robust_scaler.fit_transform(X_4)
# X_5_rob = robust_scaler.fit_transform(X_5)
# X_6_rob = robust_scaler.fit_transform(X_6)
X_7_rob = robust_scaler.fit_transform(X_7)

In [80]:
# X_1_rob_train, X_1_rob_test, y_1_rob_train, y_1_rob_test = train_test_split(X_1_rob, y_1, test_size=0.2, random_state=42)

# X_2_rob_train, X_2_rob_test, y_2_rob_train, y_2_rob_test = train_test_split(X_2_rob, y_2, test_size=0.2, random_state=42)
# X_3_rob_train, X_3_rob_test, y_3_rob_train, y_3_rob_test = train_test_split(X_3_rob, y_3, test_size=0.2, random_state=42)
# X_4_rob_train, X_4_rob_test, y_4_rob_train, y_4_rob_test = train_test_split(X_4_rob, y_4, test_size=0.2, random_state=42)
# X_5_rob_train, X_5_rob_test, y_5_rob_train, y_5_rob_test = train_test_split(X_5_rob, y_5, test_size=0.2, random_state=42)
# X_6_rob_train, X_6_rob_test, y_6_rob_train, y_6_rob_test = train_test_split(X_6_rob, y_6, test_size=0.2, random_state=42)
X_7_rob_train, X_7_rob_test, y_7_rob_train, y_7_rob_test = train_test_split(X_7_rob, y_7, test_size=0.2, random_state=42)

In [81]:
model = RandomForestRegressor()

In [82]:
# model.fit(X_1_rob_train, y_1_rob_train)
# model.fit(X_2_rob_train, y_2_rob_train)
# model.fit(X_3_rob_train, y_3_rob_train)
# model.fit(X_4_rob_train, y_4_rob_train)
# model.fit(X_5_rob_train, y_5_rob_train)
# model.fit(X_6_rob_train, y_6_rob_train)
model.fit(X_7_rob_train, y_7_rob_train)

RandomForestRegressor()

In [83]:
# pred_1_rob = model.predict(X_1_rob_test)
# pred_2_rob = model.predict(X_2_rob_test)
# pred_3_rob = model.predict(X_3_rob_test)
# pred_4_rob = model.predict(X_4_rob_test)
# pred_5_rob = model.predict(X_5_rob_test)
# pred_6_rob = model.predict(X_6_rob_test)
pred_7_rob = model.predict(X_7_rob_test)

In [84]:
# rmse_1_rob = mean_squared_error(y_1_rob_test, pred_1_rob, squared=False)
# rmse_2_rob = mean_squared_error(y_2_rob_test, pred_2_rob, squared=False)
# rmse_3_rob = mean_squared_error(y_3_rob_test, pred_3_rob, squared=False)
# rmse_4_rob = mean_squared_error(y_4_rob_test, pred_4_rob, squared=False)
# rmse_5_rob = mean_squared_error(y_5_rob_test, pred_5_rob, squared=False)
# rmse_6_rob = mean_squared_error(y_6_rob_test, pred_6_rob, squared=False)
rmse_7_rob = mean_squared_error(y_7_rob_test, pred_7_rob, squared=False)

C:\Users\tete1\miniconda3\envs\jupyter_env\lib\site-packages\sklearn\metrics\_regression.py:483: FutureWarning: 'squared' is deprecated in version 1.4 and will be removed in 1.6. To calculate the root mean squared error, use the function'root_mean_squared_error'.
  warnings.warn(


In [85]:
print(rmse_1_rob)
print(rmse_2_rob)
print(rmse_3_rob)
print(rmse_4_rob)
print(rmse_5_rob)
print(rmse_6_rob)
print(rmse_7_rob)

634.4082896385572
549.0708993453093
637.3408548852266
562.2912438468542
624.5288474670245
562.641901251339
553.2234249191879


## 🏅 Winner: 2do diamonds RSME = 549.07

# Now: Diamonds_test + NO Scaler

In [119]:
diamonds_test = pd.read_csv('./data/diamonds_test.csv')

In [120]:
diamonds_test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13485 entries, 0 to 13484
Data columns (total 11 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   id       13485 non-null  int64  
 1   carat    13485 non-null  float64
 2   cut      13485 non-null  object 
 3   color    13485 non-null  object 
 4   clarity  13485 non-null  object 
 5   depth    13485 non-null  float64
 6   table    13485 non-null  float64
 7   x        13485 non-null  float64
 8   y        13485 non-null  float64
 9   z        13485 non-null  float64
 10  city     13485 non-null  object 
dtypes: float64(6), int64(1), object(4)
memory usage: 1.1+ MB


In [87]:
# diamonds_test.drop(diamonds_test[(diamonds_test['x'] == 0) | (diamonds_test['y'] == 0) | (diamonds_test['z'] == 0)].index, inplace=True)

In [121]:
diamonds_test.drop('depth', axis=1, inplace=True)

In [122]:
diamonds_test.drop('city', axis=1, inplace=True)

In [123]:
cut_map = {'Fair': 1,
           'Good': 2,
           'Very Good': 3,
           'Premium': 4,
           'Ideal': 5}

In [124]:
diamonds_test['cut'] = diamonds_test['cut'].map(cut_map)

In [125]:
color_map = {'J': 1,
             'I': 2,
             'H': 3,
             'G': 4,
             'F': 5,
             'E': 6,
             'D': 7}

In [126]:
diamonds_test['color'] = diamonds_test['color'].map(color_map)

In [127]:
clarity_map = {'I1': 1,
               'SI2': 2,
               'SI1': 3,
               'VS2': 4,
               'VS1': 5,
               'VVS2': 6,
               'VVS1': 7,
               'IF': 8}

In [128]:
diamonds_test['clarity'] = diamonds_test['clarity'].map(clarity_map)

In [129]:
diamonds_test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13485 entries, 0 to 13484
Data columns (total 9 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   id       13485 non-null  int64  
 1   carat    13485 non-null  float64
 2   cut      13485 non-null  int64  
 3   color    13485 non-null  int64  
 4   clarity  13485 non-null  int64  
 5   table    13485 non-null  float64
 6   x        13485 non-null  float64
 7   y        13485 non-null  float64
 8   z        13485 non-null  float64
dtypes: float64(5), int64(4)
memory usage: 948.3 KB


In [130]:
diamonds_test.drop('id', axis=1, inplace=True)

In [101]:
diamonds_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40439 entries, 0 to 40438
Data columns (total 9 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   price    40439 non-null  int64  
 1   carat    40439 non-null  float64
 2   table    40439 non-null  float64
 3   x        40439 non-null  float64
 4   y        40439 non-null  float64
 5   z        40439 non-null  float64
 6   cut      40439 non-null  int64  
 7   color    40439 non-null  int64  
 8   clarity  40439 non-null  int64  
dtypes: float64(5), int64(4)
memory usage: 2.8 MB


In [102]:
diamonds_test.head()

carat  cut  color  clarity  table     x     y     z
0   0.79    3      5        3   60.0  5.82  5.89  3.67
1   1.20    5      1        5   57.0  6.81  6.89  4.18
2   1.57    4      3        3   61.0  7.38  7.32  4.57
3   0.90    3      5        3   54.0  6.09  6.13  3.90
4   0.50    3      5        5   58.0  5.05  5.09  3.19

In [131]:
diamonds_test = diamonds_test[['carat', 'table', 'x', 'y', 'z', 'cut', 'color', 'clarity']]

In [104]:
diamonds_test.head()

carat  table     x     y     z  cut  color  clarity
0   0.79   60.0  5.82  5.89  3.67    3      5        3
1   1.20   57.0  6.81  6.89  4.18    5      1        5
2   1.57   61.0  7.38  7.32  4.57    4      3        3
3   0.90   54.0  6.09  6.13  3.90    3      5        3
4   0.50   58.0  5.05  5.09  3.19    3      5        5

In [105]:
model = RandomForestRegressor()

In [132]:
pred_test = model.predict(diamonds_test)

In [133]:
pred_test

array([3001.42, 5339.79, 9279.53, ..., 2853.74, 2186.34,  852.23])

In [134]:
pred_test = pd.DataFrame(pred_test, columns=['price'])

In [135]:
pred_test['id'] = range(0, len(pred_test))

In [136]:
pred_test = pred_test[['id', 'price']]

In [137]:
pred_test

id    price
0          0  3001.42
1          1  5339.79
2          2  9279.53
3          3  4154.87
4          4  1684.86
...      ...      ...
13480  13480  1875.26
13481  13481  2518.46
13482  13482  2853.74
13483  13483  2186.34
13484  13484   852.23

[13485 rows x 2 columns]

In [138]:
pred_test.to_csv('./submissions/sub_RandomForestRegressor_548.csv', index=False)